framework/src/Volo.Abp.Uow/Volo/Abp/Uow/UnitOfWork.cs

```diff
@@ -126,6 +134,30 @@ public virtual async Task CompleteAsync(CancellationToken cancellationToken = de
             {
                 _isCompleting = true;
                 await SaveChangesAsync(cancellationToken);
+
+                while (LocalEvents.Any() || DistributedEvents.Any())
+                {
+                    if (LocalEvents.Any())
+                    {
+                        var localEventsToBePublished = LocalEvents.OrderBy(e => e.EventOrder).ToArray();
+                        LocalEvents.Clear();
+                        await UnitOfWorkEventPublisher.PublishLocalEventsAsync(
+                            localEventsToBePublished
+                        );
+                    }
+                    
+                    if (DistributedEvents.Any())
+                    {
+                        var distributedEventsToBePublished = DistributedEvents.OrderBy(e => e.EventOrder).ToArray();
+                        DistributedEvents.Clear();
+                        await UnitOfWorkEventPublisher.PublishDistributedEventsAsync(
+                            distributedEventsToBePublished
+                        );
+                    }
+
+                    await SaveChangesAsync(cancellationToken);
+                }
+                
```

Comment:
We will delete the cache in the event of the entity, but the `considerUow` feature will set the cache in `OnCompletedAsync`.

We also need to refactor the `considerUow` feature. Do you have any suggestions?

Reply:
I think this development doesn't effect the `considerUow` feature of the distributed cache. There is no way to guarantee to set the cache in the same transactional because database and cache are different systems.
It is better to keep it as is. You can merge this.

Reply:
1. The `considerUow` feature sets the cache after the unit of work is completed.
2. The event of an entity changed deletes the cache.

Now their order is reversed.

You can see the unit tests failed https://github.com/abpframework/abp/runs/3458051031